### PR TITLE
feat(Approval): use anchor to point directly to the form in the timeline

### PR DESCRIPTION
**Pull request by stonebuzz**
To be continued https://github.com/glpi-project/glpi/pull/14256

The idea would be (via HTML anchors) to position the user directly on the validation form  (which concerns him)  in the time line.

| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | !27061

### CODE REVIEW

**Review by cconard96 — APPROVED** [review]

File: src/Ajax.php [hunk 1]
```diff
@@ -371,7 +371,11 @@ public static function createTabs(
                      tab: index,
                      withtemplate: " . (int)($_GET['withtemplate'] ?? 0) . "
                   }
-               );
+                );
+
+                if (location.hash) {
+                    location.href = location.hash; // move to hanchor if needed
+                }
```
Comment (cedric-anne) [inline src/Ajax.php:378]: This is almost OK, but anchor following should be done only on first load. Indeed, if user goes to another tab and then go back on main tab, this will be evaluated again, and user will loose the position where it previously scrolled.

Comment (stonebuzz) [reply src/Ajax.php]: Last commit remove anchor from history to avoid reusing it

**Review by cedric-anne — APPROVED** [review]

**Review by orthagh — CHANGES_REQUESTED** [review]
I'm not sure about proposed changes, especially those in Auth classes.
I would prefer not altering this part for this kind of feature.

A quick suggestion for an alternative:

- we currently support html anchors (with href attributes on dates) for timeline items
- they don't fully work currently as timeline is loaded by ajax, the #id targeting an href is not present yet when browser loads glpi.
- you can simulate them by copying the link in a date of an timeline item and just pasting it in the same tab (without reloading, and scrolling elsewhere before)
- redirect param of index.php is still present for supporting former ways to redirect to a thing in glpi. But since years, after login you are redirected directly to the page asked. This parameter should not be used anymore for new features
- So, in conclusion, if we can by somehow (that's the tricky part) get the href anchor to work with ajax loading of timeline, the only thing left is to add a new href to the validation form and use it in notifications

**Comment by cedric-anne** [comment]
As proposed solution is not accepted, I close this PR.
